frontend/src/main/scala/bloop/dap/DebugSession.scala

```diff
@@ -113,7 +121,7 @@ final class DebugSession(
             case Success(address) =>
               super.dispatchRequest(DebugSession.toAttachRequest(requestId, address))
             case Failure(exception) =>
-              val cause = s"Could not start debuggee due to ${exception.getCause}"
```

Comment:
Was the cause not relevant?

Reply:
`cause` is a throwable which underlies the `exception`. When printed, it would append the entire stack trace instead of just the message.

It should have been `getMessage` all along, since that is what we want to show to the user.

Reply:
But `getMessage` doesn't throw the `getMessage` of the cause exception, so we could be showing only the superficial `getMessage` here, not the one that actually caused the issue?

Reply:
That is a question about building exceptions on top of other exceptions, so let me answer with an example.

Lets say, we want to parse a config file and we encounter a malformed integer "1foo2". The first exception here would be NumberFormatException with message like `invalid number "1foo2"`.
That is the underlying cause. But out parser knows that the property being deserialized is called "prop1", so we wrap the exception in another one with message: `Could not parse prop1 due to: ${underlying.getMessage}`. 

Of course, someone could have wrapped in an exception with message "parsing failed", but we cannot check that in the code and we have to trust the library authors and other developers to provide meaningful messages.

In summary, the top-level exception should contain a message with the broadest context, while the cause at the bottom should contain a very specific description of the problem but without any context.

Reply:
You can't assume the wrapped exceptions are always good, they aren't, for example I've run lately into many exceptions whose cause were NPEs but that were not properly handled by the libraries (in this case, lsp4j). I agree with you that in principle this should be fine.
  
Let's trust we won't run into these issues here for now.